pkg/adsc/adsc.go

```diff
@@ -782,7 +824,22 @@ func (a *ADSC) Wait(to time.Duration, updates ...string) ([]string, error) {
 	for {
 		select {
 		case t := <-a.Updates:
-			delete(want, t)
+			if t == "" {
+				return got, fmt.Errorf("closed")
+			}
+			toDelete := t
+			// legacy names, still used in tests.
+			switch t {
+			case ListenerType:
+				delete(want, "lds")
+			case v2.ClusterTypeV3:
+				delete(want, "cds")
+			case v2.EndpointTypeV3:
+				delete(want, "eds")
+			case routeType:
+				delete(want, "rds")
```

Comment:
@costinm  can you explain why is this change needed?

Reply:
Sync up with the V3 change and make the code less dependent of the old endpoints. 

ADSC is a client for XDS/ADS - was initially used for testing, but I'm planning to use it in agent for pulling ServiceEntry from Istiod so it can responds to DNS without a RTT. 
It needs a bit of cleanup...